packages/cadl-ranch-specs/http/payload/multipart/main.tsp

```diff
@@ -332,4 +329,132 @@ Content-Type: application/octet-stream;
+--abcde12345--
+```
+""")
+  @doc("Test content-type: multipart/form-data")
+  @post
+  @route("/check-filename-and-specific-content-type-with-httppart")
+  op fileWithHttpPartSpecificContentType(
+    @header contentType: "multipart/form-data",
+    @multipartBody body: FileWithHttpPartSpecificContentTypeRequest,
+  ): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+This case will check required content-type of file part, so expect request:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345--
+```
+""")
+  @doc("Test content-type: multipart/form-data")
+  @post
+  @route("/check-filename-and-required-content-type-with-httppart")
+  op fileWithHttpPartRequiredContentType(
+    @header contentType: "multipart/form-data",
+    @multipartBody body: FileWithHttpPartRequiredContentTypeRequest,
+  ): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+Please send request twice, first time with no content-type and second time with content-type "application/octet-stream". Expect request:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345
+```
+""")
+  @doc("Test content-type: multipart/form-data for optional content type")
+  @post
+  @route("/file-with-http-part-optional-content-type")
+  op fileWithHttpPartOptionalContentType(
+    @header contentType: "multipart/form-data",
+    @multipartBody body: FileWithHttpPartOptionalContentTypeRequest,
+  ): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+For File part, filename will not be checked but it is necessary otherwise cadl-ranch can't parse it;
+content-type will be checked with value "application/octet-stream". Expect request:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="id"
+Content-Type: text/plain
+
+123
+--abcde12345
+Content-Disposition: form-data; name="address"
+Content-Type: application/json
+
+{
+  "city": "X"
+}
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345--
+Content-Disposition: form-data; name="previousAddresses"
+Content-Type: application/json
+
+[{
+  "city": "Y"
+},{
+  "city": "Z"
+}]
+--abcde12345
+Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
+--abcde12345
+Content-Disposition: form-data; name="pictures"; filename="<any-name-is-ok>"
+Content-Type: application/octet-stream
+
+{…file content…}
```

Comment:
Preferrable to also specify e.g. 2 Png files

Reply:
updated